packages/core/src/odata/common/filter/filter-function.ts

```diff
@@ -49,14 +49,47 @@ export function filterFunction<EntityT extends EntityBase>(
   return new NumberFilterFunction(
     functionName,
     parameters,
-    returnTypeToEdmType[returnType]
+    numberReturnTypeMapping[returnType]
   );
 }
 
-const returnTypeToEdmType: MapType<
+// eslint-disable-next-line valid-jsdoc
+/**
+ * @hidden
+ */
+export function createFilterFunction<EntityT extends EntityBase>(
+  functionName: string,
+  returnType: FilterFunctionReturnType,
+  ...parameters: FilterFunctionParameterType<EntityT>[]
+):
+  | BooleanFilterFunction<EntityT>
+  | NumberFilterFunction<EntityT>
+  | StringFilterFunction<EntityT> {
+  if (returnType === 'boolean' || returnType === 'bool') {
+    return new BooleanFilterFunction(functionName, parameters);
+  }
+  if (returnType === 'string') {
+    return new StringFilterFunction(functionName, parameters);
+  }
+  return new NumberFilterFunction(
+    functionName,
+    parameters,
+    numberReturnTypeMapping[returnType]
+  );
+}
+
+const numberReturnTypeMapping: MapType<
```

Comment:
Minor: is it possible also to enforce the three keys: int,double,decimal to be there via the type definition?

Reply:
Also this could be one of the few cases where a `switch` could be nice since you could group the cases:
```
 switch (returnType) {
    case 'boolean':
    case 'bool':
      return boolStuff
    case 'int':
    case 'double':
    case 'decimal':
      return numberStuff
    case 'string':
      return stringStuff
  }
```

Reply:
I don't fully understand, can you elaborate?

Reply:
Just saying that you could use a switch there. But ifs are also fine.